x-pack/legacy/plugins/ml/public/application/jobs/new_job/common/job_creator/job_creator.ts

```diff
@@ -188,12 +193,12 @@ export class JobCreator {
     this._job_config.groups = groups;
   }
 
-  public get calendars(): string[] {
-    return this._job_config.calendars || [];
+  public get calendars(): Calendar[] {
+    return this._calendars || [];
```

Comment:
nit, `this.calendars` will not be falsy now, so this condition could be removed.
similar to the check in `_updateCalendars`

Reply:
https://github.com/elastic/kibana/pull/51281/commits/58e5372073afdd077833f5956f7df40341378eda